### PR TITLE
app/vmauth: add option to skip TLS verification

### PR DESCRIPTION
**Pull request by zekker6**
Add `tls_insecure_skip_verify` option on per-user basis which allows to disable TLS verification for all requests to backend on behalf of this user.

See: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5240

### CODE REVIEW

**Comment by f41gh7** [comment]
I think, it'd be great to add all tls related features ( ca, cert, key).

File: app/vmauth/main.go [hunk 1]
```diff
@@ -345,12 +346,7 @@ var (
 	missingRouteRequests     = metrics.NewCounter(`vmauth_http_request_errors_total{reason="missing_route"}`)
 )
 
-var (
-	transport     *http.Transport
-	transportOnce sync.Once
-)
-
-func transportInit() {
+func getTransport(skipTLSVerification bool) *http.Transport {
```
Comment (f41gh7) [inline app/vmauth/main.go:349]: This change increases resource usage and slows vmauth performance. Possible workaround:

 create an additional instance with disable tls and select transport based on enabled/disabled tls.

Comment (zekker6) [reply app/vmauth/main.go:351]: Added a simple cache to reuse already created transport with the same TLS verification option.

**Comment by zekker6** [comment]
> I think, it'd be great to add all tls related features ( ca, cert, key).

cert and key seems to be a part of mTLS support which is a part of ent version currently.

setting CA alone seems to be just the same as updating base OS CA, so I'm not sure if it makes a lot of sense to support those options atm

File: app/vmauth/main.go (hunk 1 above)
Comment (f41gh7) [reply app/vmauth/main.go:351]: I don't really like sync.Map for this case. 

 As an option. We can create http.Transport per auth config record. It'll simplify code for enterprise version with full tls options support.

**Review by hagen1778 — APPROVED** [review]
LGTM!

**Pull request merged**